src/RBTree.mo

```diff
@@ -8,80 +39,286 @@ import O "Order";
 
 module {
 
-  /// Node color: red or black.
+  /// Node color: Either red (`#R`) or black (`#B`).
   public type Color = { #R; #B };
 
-  /// Ordered, (red-black) tree of entries.
-  public type Tree<X, Y> = {
-    #node : (Color, Tree<X, Y>, (X, ?Y), Tree<X, Y>);
+  /// Red-black tree of nodes with key-value entries, ordered by the keys.
+  /// The keys have the generic type `K` and the values the generic type `V`.
+  /// Leaves are considered implicitly black.
+  public type Tree<K, V> = {
+    #node : (Color, Tree<K, V>, (K, ?V), Tree<K, V>);
     #leaf
   };
 
-  /// Create an order map from an order function for its keys.
-  public class RBTree<X, Y>(compareTo : (X, X) -> O.Order) {
+  /// A map from keys of type `K` to values of type `V` implemented as a red-black tree.
+  /// The entries of key-value pairs are ordered by `compareTo` function applied to the keys.
+  ///
+  /// The class enables imperative usage in object-oriented-style.
+  /// However, internally, the class uses a functional implementation.
+  ///
+  /// The `compareTo` function should implement a consistent total order among all possible values of `K` and
+  /// for efficiency, only involves `O(1)` runtime costs without space allocation.
+  ///
+  /// Example:
+  /// ```motoko name=initialize
+  /// import RBTree "mo:base/RBTree";
+  /// import Nat "mo:base/Nat";
+  ///
+  /// let tree = RBTree.RBTree<Nat, Text>(Nat.compare); // Create a map of `Nat` to `Text` using the `Nat.compare` order
+  /// ```
+  ///
+  /// Costs of instantiation (only empty tree):
+  /// Runtime: `O(1)`.
+  /// Space: `O(1)`.
+  public class RBTree<K, V>(compareTo : (K, K) -> O.Order) {
```

Comment:
```suggestion
  public class RBTree<K, V>(compareTo: (K, K) -> O.Order) {
```
Since you are nicely renaming stuff, maybe we should also rename `compareTo` to `compare` unless that clashes somewhere. Only if you feel like it.

Reply:
Indeed, this improves it. I changed it. (The renaming is merely internal and does not clash with anything external.)